soc/riscv/riscv-privilege/hpmicro/soc.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2022 hpmicro
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+#include <init.h>
+extern void board_init_ahb(void);
```

Comment:
why do we need board specific hooks? can you please explain?

Reply:
ping, no response here.

Reply:
Initialized the soc clock and placed it under the board in hpm_sdk. I wanted to reuse the code so I called it directly.